.github/workflows/release-python.yml

```diff
@@ -0,0 +1,150 @@
+name: Python Wheels
+
+on:
+  push:
+    tags:
+      - "[0-9]+.[0-9]+.[0-9]+"
+      - "[0-9]+.[0-9]+.[0-9]+.post[0-9]+"
+      - "[0-9]+.[0-9]+.[0-9]+[a-b][0-9]+"
+      - "[0-9]+.[0-9]+.[0-9]+rc[0-9]+"
+  workflow_dispatch:
+
+concurrency:
+  group: wheels-${{ github.ref }}
+  cancel-in-progress: true
+
+defaults:
+  run:
+    shell: bash -eux {0}
+
+jobs:
+  build_wheels:
+    name: Build wheel for ${{ matrix.python }}-${{ matrix.buildplat[1] }}
+    runs-on: ${{ matrix.buildplat[0] }}
+    strategy:
+      # Ensure that a wheel builder finishes even if another fails
+      fail-fast: false
+      matrix:
+        # Github Actions doesn't support pairing matrix values together, let's improvise
+        # https://github.com/github/feedback/discussions/7835#discussioncomment-1769026
+        buildplat:
+        - [ubuntu-20.04, manylinux_x86_64]
+        - [ubuntu-20.04, manylinux_aarch64]
+        - [ubuntu-20.04, manylinux_ppc64le]
+        - [ubuntu-20.04, manylinux_s390x]
+        - [ubuntu-20.04, manylinux_i686]
+        - [macos-11, macosx_*]
+        - [windows-2019, win_amd64]
+        - [windows-2019, win32]
+        python: ["cp37", "cp38", "cp39", "cp310", "cp311", "cp312"]
+
+    steps:
+      - name: Checkout pymongoarrow
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Set up python version
+        run: |
+          export PYTHON_VERSION=$(sed 's/^cp3/3./' <<< ${{ matrix.python }} )
+          echo "PYTHON_VERSION=$PYTHON_VERSION" >> $GITHUB_ENV
+
+      - uses: actions/setup-python@v4
+        with:
+          python-version: ${{env.PYTHON_VERSION}}
+          cache: 'pip'
+          cache-dependency-path: 'pyproject.toml'
+          allow-prereleases: true
+
+      - name: Set up QEMU
+        if: runner.os == 'Linux'
+        uses: docker/setup-qemu-action@v2
+        with:
+          platforms: all
+
+      - name: Install cibuildwheel
+        run: python -m pip install "cibuildwheel>=2.4,<3"
+
+      - name: Build MacOS Py37 Wheel
+        if: ${{ matrix.python == 'cp37' && matrix.buildplat[0] == 'macos-11' }}
+        env:
+          CIBW_BUILD: cp37-macosx_x86_64
+          CIBW_ARCHS: x86_64
+          CIBW_TEST_COMMAND: "python {project}/tools/fail_if_no_c.py"
+        run: python -m cibuildwheel --output-dir wheelhouse
+
+      - name: Build wheels
+        if: ${{ matrix.python != 'cp37' || matrix.buildplat[0] != 'macos-11' }}
```

Comment:
Could you add a comment explaining why 3.7 and macos 11 skip this step?

Reply:
Done